addons/mail/static/src/models/composer/composer.js

```diff
@@ -82,6 +82,12 @@ function factory(dependencies) {
             });
         }
 
+        getMessagePartners() {
+            const mentionned = this.mentionedPartners.map(partner => partner.id);
+            const suggestedPartners = this.thread.suggestedPartners.filter(partner => partner.checked).map(partner => partner.id);
+            return _.union(mentionned, suggestedPartners);
+        }
```

Comment:
- should be a computed field
- "message partners" is a weak naming. "needaction partners" is better, considering the notion of "needaction".
- do not use `_`, prefer native solutions instead. Recently had an issue with external livechat not working on some websites because they had global `_` used for lodash, while we use it for underscorejs... yeah it break stuffs.

Reply:
> should be a computed field

Why ? It's only used inside the `Composer` class. 
Could be private.

> "message partners" is a weak naming. "needaction partners" is better, considering the notion of "needaction".

I fail to see where you see a "needaction" in this context. 

> do not use _, prefer native solutions instead. Recently had an issue with external livechat not working on some websites because they had global _ used for lodash, while we use it for underscorejs... yeah it break stuffs.

Please provide a "native" function that merge 2 arrays without duplicate entry.

Reply:
There is no native function that exactly does this. You have to it in in a few expressions, like `[...new Set(...[])]` (insert array items in set, then back in array, so that each element is unique).

Reply:
You want to change a perfectly readable and understandable (for junior and senior) code to some non-sens Javascript trick ? I don't really like this approach.
Or you have to wrap the trick inside an understandable function.

Reply:
underscorejs has many drawbacks: versioning of lib, conflict with other libs (livechat external lib doesn't work in some websites, like shopify, because of that), silent errors (encourages bad programming habits), etc.

Reply:
There's no trick here, spread operator is a basic operator of modern JS and so is the Set collection.

`list(set(...))` is done plenty of time in Python as well to get unique values.

Reply:
Note that depending on context, the convert from Set back to list is often not needed, but is done for bad reasons, such as the rest of the code not handling Set correctly.

If the method is expected to return a list of unique elements, it would be better if it directly returned the Set, and its caller handles the Set correctly.

Reply:
Intent is to make this as a field, so it can just use `Set()` from recent changes.

Reply:
Well I disagree with that. But after all, if you want a `Set`, so be it.